ckan/config/middleware/__init__.py

```diff
@@ -34,3 +39,37 @@ def make_app(conf: Union[Config, CKANConfig]) -> CKANApp:
     _internal_test_request_context = flask_app._wsgi_app.test_request_context()
 
     return flask_app
+
+
+class CKANSecureCookieSessionInterface(SecureCookieSessionInterface):
+    """Flask cookie-based sessions with expiration support.
+    """
+
+    def __init__(self, app: CKANApp):
+        pass
+
+    def open_session(self, app: CKANApp, request: Request):
+        session = super().open_session(app, request)
+        if session:
+            # Cookie-based sessions expire with the browser's session. The line
+            # below changes this behavior, extending session's lifetime by
+            # `PERMANENT_SESSION_LIFETIME` seconds. `SESSION_PERMANENT` option
+            # is used as indicator of permanent sessions by flask-session
+            # package, so we also should rely on it, for predictability.
```

Comment:
I would move this as docstring of the class.

Also, why are we moving out of the default behavior? Just for predictability or is there any CKAN related feature that requires this? If so, it would be nice to document it.

Reply:
I prefer to keep this explanation directly around this line because it doesn't change the general behavior of the session storage, just adds a small feature. And usually, features are described in the place, they are implemented. But yes, it has sense to mention this in the class documentation as well, to explain the puprose of the whole custom class

> why are we moving out of the default behavior
I think you misunderstood the idea. We are not moving out of the default behavior. We are adding **a new feature** and immitating default behavior for it. In the comment I explained exactly it:
1. cookie-session expires with browser session out of the box
2. we change this behavior in the line below the comment
3. To be predictable(i.e. be similar to other session backends), we rely on `SESSION_PERMANENT` flag

> Also, why we need custom logic for this
Once again, misundestanding? You pointing at documentation of `flask-session` but asking about `flask`'s core functionality. `flask-session` describes only it's own behavior. Whatever it says about `PERMANENT_SESSION_LIFETIME` is not necessarily true for core `flask`. And that's exactly why we need custom logic - to immitate behavior of `flask-session` for `flask`'s core cookie-based session